### PR TITLE
Refactored and made the home page better designed, data aware and responsive

### PR DESCRIPTION
**Pull request by mayor04**
The widget has been updated to be data aware rather than being hardcoded, A more manageable text style has been added to enable easy comprehension. I introduces the svg icons to make icons sharper and follow the required style guide. Lastly the home screen UI has been totally changed to mimic the look and feel provided by the designs.

### CODE REVIEW

File: lib/ui/shared/text_styles.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:hng/ui/shared/colors.dart';
+
+class ZuriTextStyle {
```
Comment (isaacadariku) [inline lib/ui/shared/text_styles.dart:4]: Also can we just make this not have a class and have all those properties be constant.
It's better to just use 
```
Text(
style: smallLight,
)
```

Than having to

`ZuriTextStyle.smallLight`

File: lib/ui/view/nav_bar/nav_bar_viewmodel.dart [hunk 2]
```diff
@@ -1,7 +1,9 @@
+import 'package:flutter/material.dart';
 import 'package:stacked/stacked.dart';
 
 class NavBarViewModel extends IndexTrackingViewModel {
   static final NavBarViewModel instance = NavBarViewModel._internal();
+  List<BottomNavigationBarItem> bottomNavList = [];
```
Comment (isaacadariku) [inline lib/ui/view/nav_bar/nav_bar_viewmodel.dart:6]: I don't think there is need refactoring this to be in the viewmodel, It has already break a rule, that ensure material or cupertino widget is not imported into the VM. Please make the changes in the view.

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Great work, as soon you get the request made, kindly let me know

File: lib/ui/shared/text_field.dart [hunk 3]
```diff
@@ -9,38 +10,48 @@ import 'package:flutter/material.dart';
...
-          fontSize: 14,
+          fontSize: 16,
           color: Colors.black45,
         ),
         contentPadding: EdgeInsets.fromLTRB(16, 5, 16, 5),
       ),
     );
   }
+
+  border({Color? color}) {
```
[the remark this answers is not in the document]
Comment (mayor04) [reply lib/ui/shared/text_field.dart:47]: This text field is not used in any part of the app, we've created other text field

File: lib/ui/view/nav_bar/nav_bar_viewmodel.dart (hunk 2 above)
Comment (mayor04) [reply lib/ui/view/nav_bar/nav_bar_viewmodel.dart:6]: Done

File: lib/ui/nav_pages/home_page/widgets/home_expanded.dart [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import 'package:hng/ui/nav_pages/home_page/widgets/home_list_items.dart';
+import 'package:hng/ui/shared/colors.dart';
+import 'package:hng/ui/shared/text_styles.dart';
+import 'package:hng/utilities/constants.dart';
+import 'package:stacked/stacked.dart';
+
+import '../home_page_viewmodel.dart';
+import '../home_item_model.dart';
+
+enum HomeListType {
```
[the remark this answers is not in the document]
Comment (mayor04) [reply lib/ui/nav_pages/home_page/widgets/home_expanded.dart:11]: Done

File: lib/ui/nav_pages/home_page/widgets/home_expanded.dart [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    });
+
+    if (title == 'Channels') {
+      expansionList.add(AddChannelsTextAndIcon());
+    }
+
+    return expansionList;
+  }
+
+  Widget expansionTile({
```
[the remark this answers is not in the document]
Comment (mayor04) [reply lib/ui/nav_pages/home_page/widgets/home_expanded.dart:62]: The widget is not shared anywhere so I extracted it to the same folder

[the remark this answers is not in the document]
Comment (mayor04) [reply lib/ui/nav_pages/home_page/widgets/home_expanded.dart:62]: Done

File: lib/ui/shared/text_styles.dart (hunk 1 above)
Comment (mayor04) [reply lib/ui/shared/text_styles.dart:4]: This was a collective refactor 
The main reason why we went with this format is for font style flexibility(change the typeface and reflects everywhere) and ease to use 
Changing the styles in the ZuriTextStyle to constant would only allow constant values to be inserted which might not be compatible with our colours. I would need to contact the team lead so as to do project-wide refactoring.

**Review by isaacadariku — APPROVED** [review]
Nice one!

**Pull request merged**